py/torch_tensorrt/dynamo/runtime/_PythonTorchTensorRTModule.py

```diff
@@ -104,16 +106,138 @@ def __init__(
+            )
+        else:
+            raise RuntimeError(
+                "Invalid usage. both budget_bytes and weight_streaming_percent are None"
+            )
+
+        budget_bytes = int(budget_bytes_to_set)
+        self.engine.weight_streaming_budget_v2 = budget_bytes
+        if self.engine.weight_streaming_budget_v2 != budget_bytes:
+            RuntimeError(f"Failed to set weight streaming budget to {budget_bytes}!")
+        if budget_bytes == self.engine.streamable_weights_size:
+            logger.debug("Weight streaming is disabled.")
+        else:
+            logger.debug(
+                f"Weight streaming is enabled with a memory budget of {budget_bytes} bytes."
+            )
+            self.weight_streaming_budget = budget_bytes
+            self.streamable_weights_size = self.engine.streamable_weights_size
+
+    def set_weight_streaming_budget_v1(
+        self,
+        budget_bytes: Union[None, int],
+        weight_streaming_percent: Union[None, int],
+    ) -> None:
+        if budget_bytes == 0:
+            # TRT to choose to weight stream automatically
+            budget_bytes = -1
+        elif weight_streaming_percent is not None:
+            min_budget = self.engine.minimum_weight_streaming_budget
+            max_budget = self.engine.streamable_weights_size
+            requested_budget_bytes = (1 - weight_streaming_percent / 100.0) * (
+                max_budget - min_budget
+            ) + min_budget
+            budget_bytes = int(requested_budget_bytes)
+        if budget_bytes is not None:
+            self.engine.weight_streaming_budget = budget_bytes
+            if self.engine.weight_streaming_budget != budget_bytes:
+                RuntimeError(
+                    f"Failed to set weight streaming budget to {budget_bytes}!"
+                )
+            if budget_bytes == -1:
+                logger.debug(
+                    "Weight streaming is enabled with TensorRT automatically determining the budget."
+                )
+            else:
+                logger.debug(
+                    f"Weight streaming is enabled with a memory budget of {budget_bytes} bytes."
+                )
+                self.weight_streaming_budget = budget_bytes
+                self.streamable_weights_size = (
+                    self.engine.minimum_weight_streaming_budget
+                )
+
+    def reset_context(self):
+        # Context needs to recreated in next because if modified runtime configuration
+        if self.context != None:
+            del self.context
+            self.context = None
+
+    def lazy_init_context(self):
+        # Lazy creation of context when it is actually needed at forward() or enable_profiling()
+        assert self.engine, f"Context is used before setting up the engine"
+
+        if self.context is None:
+            self.context = self.engine.create_execution_context()
```

Comment:
we already have a setup engine function, not sure why we need to handle this at exec time?